iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -168,6 +171,11 @@ LogicalResult TileSizeCalculator::inferTileAndWorkgroupSize(
     workgroupSize = {8, 8, 1};
     return success();
   }
+  if (opInfo & OpInfo::BatchMatmul) {
+    tileSizes = {2, 8, 8, 4};
+    workgroupSize = {2, 8, 8};
```

Comment:
Should this be in the reverse of tileSizes and resize to 3, ie, {4, 8, 8}?

Reply:
It should be `{8, 8, 2}`. The tile sizes `{2, 8, 8, 4}` are for the batch-loop, M, N and K loops. The batch, M and N loops are distributed. along z-dimension, y-dimension and x-dimension. K loop is sequential. So even though it is tiled, it isnt distributed. Finally, the workgroup size is described as `x-dimension, y-dimension, z-dimension`. So that gives you `{8, 8, 2}`. 
(I am surprised this gave correct results. I think it just happened to give the right answer for the tests, but in general it would fail.)
Side Note: The `k` loop tiling is mostly for workgroup memory promotion. If we are not doing that, we dont really need to tile the `k`-loop. For now we are doing this, but there is a case to not do that by default.

Reply:
Done.

I'm curious that if the workgroupSize of Matmul should be {4, 8, 8}? I guess the reason is we only consider parallel loops, so the "tileSize" in Matmul Op (that used in workgroupSize) is {8, 8}. Therefor, the workgroupSize is {8, 8, 1} instead. Is it correct?